crates/runtime/src/dataconnector.rs

```diff
@@ -393,35 +392,31 @@ impl<T: ListingTableConnector + Display> DataConnector for T {
 
         let url = self
             .get_object_store_url(dataset)
-            .context(InvalidConfigurationSnafu {
+            .context(ListingTableInternalSnafu {
                 dataconnector: format!("{__self}"),
-                message: "Unable to parse URL",
             })?;
 
         let table_path =
             ListingTableUrl::parse(url)
                 .boxed()
-                .context(InvalidConfigurationSnafu {
+                .context(ListingTableInternalSnafu {
                     dataconnector: format!("{__self}"),
-                    message: "Unable to parse URL",
                 })?;
 
         let (file_format, extension) = self
             .get_file_format_and_extension()
             .map_err(Into::into)
-            .context(InvalidConfigurationSnafu {
+            .context(ListingTableInternalSnafu {
```

Comment:
`get_file_format_and_extension` already returns DataConnectorError, why we need map and context here?